ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -169,7 +169,9 @@ def get_timestamp_field(self) -> str:
             return f"{self._trunc_func}(toDateTime({self.EVENT_TABLE_ALIAS}.timestamp, %(timezone)s)) AS event_date"
 
     def _determine_should_join_distinct_ids(self) -> None:
-        if self._filter.aggregation_group_type_index is not None or self._aggregate_users_by_distinct_id:
+        if (
+            self._filter.aggregation_group_type_index is not None or self._aggregate_users_by_distinct_id
+        ) and not self._column_optimizer.is_using_cohort_propertes:
```

Comment:
This condition is a bit troubling because I think we're overriding person_ids in some cases to make queries faster. But, if we're honoring cohort filtering, right now the cohort queries need a person_id to filter on so this needs to be handled, either 
1. we leave it like what I just did where persons will still be joined if cohort properties exist
2. update cohort filter condition to work on distinct_id instead of requiring person_id